Libraries/LibCore/DateTime.cpp

```diff
@@ -51,9 +50,15 @@ DateTime DateTime::from_timestamp(time_t timestamp)
     return dt;
 }
 
-String DateTime::to_string() const
+String DateTime::to_string(const String& format) const
 {
-    return String::format("%04u-%02u-%02u %02u:%02u:%02u", m_year, m_month, m_day, m_hour, m_minute, m_second);
+    struct tm tm;
+    localtime_r(&m_timestamp, &tm);
+    char buff[256];
```

Comment:
Why 256? This seems very arbitrary. Can we organize the code to allow any length?

Reply:
Also please capitalize commit messages :)

Reply:
As  we must pass a buffer size to the C function `strftime()`, if we want to allow arbitrary size, we 
can't call it directly,  and have to copy large parts of code from `strftime()`.